gmail/src/services/odoo_auth.ts

```diff
@@ -0,0 +1,133 @@
+import { ODOO_AUTH_URLS } from "../const";
+import { postJsonRpc } from "../utils/http";
+
+const errorPage = `
+<html>
+    <style>
+        .alert {
+            color: #721c24;
+            background-color: #f5c6cb;
+            padding: 20px;
+            max-width: 1000px;
+            margin: auto;
+            text-align: center;
+            font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol"
+        }
+        img {
+            max-width: 300px;
+            margin-left: calc(50% - 150px);
+            margin-bottom: 50px;
+        }
+        hr {
+            border- color: #721c24;
+        }
+    </style>
+    <img src="https://raw.githubusercontent.com/odoo/mail-client-extensions/master/outlook/assets/odoo-full.png">
+    <div class="alert">__ERROR_MESSAGE__</div>
+</html>`;
+
+/**
+ * Callback function called during the OAuth authentication process.
+ *
+ * 1. The user click on the "Login button"
+ *    We generate a state token (for this function)
+ * 2. The user is redirected to Odoo and enter his login / password
+ * 3. Then the user is redirected to the Google App-Script
+ * 4. Thanks the the state token, the function "odooAuthCallback" is called with the auth code
+ * 5. The auth code is exchanged for an access token with a RPC call
+ */
+function odooAuthCallback(callbackRequest: any) {
+    Logger.log("Run authcallback");
```

Comment:
How do those work? It logs stuff in the JS console?

Reply:
It log in the "google console" (on the google project you can see all the logs of the application)

Reply:
And do we actually want to see hundreds of those "Run authcallback" without any context on our Google Project?
Does it have any use?